pac4j-saml/src/main/java/org/pac4j/saml/util/Configuration.java

```diff
@@ -46,55 +55,97 @@ private Configuration() {
         bootstrap();
     }
 
+    private static int compareManagers(Object obj1, Object obj2) {
+        int p1 = 100;
+        int p2 = 100;
+        Priority p1a = obj1.getClass().getAnnotation(Priority.class);
+        if (p1a != null) {
+            p1 = p1a.value();
+        }
+        Priority p2a = obj2.getClass().getAnnotation(Priority.class);
+        if (p2a != null) {
+            p2 = p2a.value();
+        }
+        if (p1 < p2) {
+            return -1;
+        } else if (p1 > p2) {
+            return 1;
+        } else {
+            return obj2.getClass().getSimpleName().compareTo(obj1.getClass().getSimpleName());
+        }
+    }
+
     private static void bootstrap() {
-        parserPool = new BasicParserPool();
-        parserPool.setMaxPoolSize(100);
-        parserPool.setCoalescing(true);
-        parserPool.setIgnoreComments(true);
-        parserPool.setNamespaceAware(true);
-        parserPool.setExpandEntityReferences(false);
-        parserPool.setXincludeAware(false);
-        parserPool.setIgnoreElementContentWhitespace(true);
-
-        final Map<String, Object> builderAttributes = new HashMap<String, Object>();
-        parserPool.setBuilderAttributes(builderAttributes);
-
-        final Map<String, Boolean> features = new HashMap<>();
-        features.put("http://apache.org/xml/features/disallow-doctype-decl", Boolean.TRUE);
-        features.put("http://apache.org/xml/features/validation/schema/normalized-value", Boolean.FALSE);
-        features.put("http://javax.xml.XMLConstants/feature/secure-processing", Boolean.TRUE);
-        features.put("http://xml.org/sax/features/external-general-entities", Boolean.FALSE);
-        features.put("http://xml.org/sax/features/external-parameter-entities", Boolean.FALSE);
-
-        parserPool.setBuilderFeatures(features);
+        ServiceLoader<ConfigurationManager> configurationManagers = ServiceLoader.load(ConfigurationManager.class);
+        List<ConfigurationManager> configurationManagerList = new ArrayList();
+        configurationManagers.forEach(configurationManagerList::add);
+        if (configurationManagerList.size() > 0) {
+            configurationManagerList.sort(Configuration::compareManagers);
+            configurationManagerList.get(0).configure();
+        }
 
-        try {
-            parserPool.initialize();
-        } catch (final ComponentInitializationException e) {
-            throw new RuntimeException("Exception initializing parserPool", e);
+        for (ConfigurationManager configurationManager : configurationManagers) {
+            configurationManager.configure();
         }
+    }
 
+    public static class DefaultConfigurationManager implements ConfigurationManager {
```

Comment:
Unless there is a specific reason, I'd prefer to have this class as a separate one.